From 4774150fcdb2872731b224a6c534621dae3772c1 Mon Sep 17 00:00:00 2001
From: Ivan Mironov <mironov.ivan@gmail.com>
Date: Tue, 25 Aug 2026 04:03:24 +0200
Subject: [PATCH 1/4] Fix indigo_ccd_rpi
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This fixes following compilation errors:
	...
	g++ -g -fPIC -O3 -isystem/builddir/build/BUILD/indigo-3.0.4-build/indigo-3.0-4/indigo_libs -I/builddir/build/BUILD/indigo-3.0.4-build/indigo-3.0-4/indigo_drivers -I/builddir/build/BUILD/indigo-3.0.4-build/indigo-3.0-4/indigo_linux_drivers -I/builddir/build/BUILD/indigo-3.0.4-build/indigo-3.0-4/build/include -std=gnu++11 -pthread -DINDIGO_LINUX -I/usr/include/libraw -g -fPIC -O3 -Wall -I../../indigo_libs -I../../indigo_drivers -I/usr/include/libcamera -I/builddir/build/BUILD/indigo-3.0.4-build/indigo-3.0-4/build/include -std=gnu++17 -pthread -DINDIGO_LINUX   -c -o indigo_ccd_rpi.o indigo_ccd_rpi.cpp
	indigo_ccd_rpi.cpp: In function ‘void exposure_timer_callback(indigo_device*)’:
	indigo_ccd_rpi.cpp:971:56: warning: ‘?:’ using integer constants in boolean context [-Wint-in-bool-context]
	  971 |                 ready = pRPiCamera->IsRequestCreated() ? pRPiCamera->IsAnyRequestComplete() : -1;
		  |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
	indigo_ccd_rpi.cpp: In function ‘indigo_result ccd_attach(indigo_device*)’:
	indigo_ccd_rpi.cpp:1059:17: warning: statement has no effect [-Wunused-value]
	 1059 |                 INDIGO_FAILED;
		  |                 ^~~~~~~~~~~~~
	indigo_ccd_rpi.cpp: In function ‘indigo_result ccd_change_property(indigo_device*, indigo_client*, indigo_property*)’:
	indigo_ccd_rpi.cpp:1562:33: warning: comparison of integer expressions of different signedness: ‘unsigned int’ and ‘int’ [-Wsign-compare]
	 1562 |                         if (bin == horizontal_bin && !strncmp(item->name, selected_name, 5)) {
		  |                             ~~~~^~~~~~~~~~~~~~~~~
	indigo_ccd_rpi.cpp: In function ‘indigo_result ccd_enumerate_properties(indigo_device*, indigo_client*, indigo_property*)’:
	indigo_ccd_rpi.cpp:1667:17: error: ‘indigo_define_matching_property’ was not declared in this scope; did you mean ‘indigo_device_change_property’?
	 1667 |                 indigo_define_matching_property(RPI_AF_PROPERTY);
		  |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
		  |                 indigo_device_change_property
	...
	gcc -g -fPIC -O3 -isystem/builddir/build/BUILD/indigo-3.0.4-build/indigo-3.0-4/indigo_libs -I/builddir/build/BUILD/indigo-3.0.4-build/indigo-3.0-4/indigo_drivers -I/builddir/build/BUILD/indigo-3.0.4-build/indigo-3.0-4/indigo_linux_drivers -I/builddir/build/BUILD/indigo-3.0.4-build/indigo-3.0-4/build/include -std=gnu11 -pthread -DINDIGO_LINUX -DRPI_MANAGEMENT -I/usr/include/libraw   -c -o indigo_ccd_rpi_main.o indigo_ccd_rpi_main.c
	indigo_ccd_rpi_main.c: In function ‘main’:
	indigo_ccd_rpi_main.c:33:66: error: passing argument 2 of ‘indigo_xml_device_adapter’ makes pointer from integer without a cast [-Wint-conversion]
	   33 |   indigo_client *protocol_adapter = indigo_xml_device_adapter(0, 1);
		  |                                                                  ^
		  |                                                                  |
		  |                                                                  int
	In file included from indigo_ccd_rpi_main.c:27:
	/builddir/build/BUILD/indigo-3.0.4-build/indigo-3.0-4/indigo_libs/indigo/indigo_driver_xml.h:48:101: note: expected ‘indigo_uni_handle *’ but argument is of type ‘int’
	   48 | INDIGO_EXTERN indigo_client *indigo_xml_device_adapter(indigo_uni_handle *input, indigo_uni_handle *output);
		  |                                                                                  ~~~~~~~~~~~~~~~~~~~^~~~~~
	...
---
 indigo_optional_drivers/ccd_rpi/indigo_ccd_rpi.cpp    | 2 +-
 indigo_optional_drivers/ccd_rpi/indigo_ccd_rpi_main.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/indigo_optional_drivers/ccd_rpi/indigo_ccd_rpi.cpp b/indigo_optional_drivers/ccd_rpi/indigo_ccd_rpi.cpp
index 738b6e08142d..370b29877af5 100644
--- a/indigo_optional_drivers/ccd_rpi/indigo_ccd_rpi.cpp
+++ b/indigo_optional_drivers/ccd_rpi/indigo_ccd_rpi.cpp
@@ -1664,7 +1664,7 @@ static indigo_result ccd_enumerate_properties(indigo_device *device, indigo_clie
 	assert(DEVICE_CONTEXT != NULL);
 	if (IS_CONNECTED) {
 
-		indigo_define_matching_property(RPI_AF_PROPERTY);
+		INDIGO_DEFINE_MATCHING_PROPERTY(RPI_AF_PROPERTY);
 
 		if (X_CCD_ADVANCED_PROPERTY && indigo_property_match(X_CCD_ADVANCED_PROPERTY, property)) {
 			indigo_define_property(device, X_CCD_ADVANCED_PROPERTY, NULL);
diff --git a/indigo_optional_drivers/ccd_rpi/indigo_ccd_rpi_main.c b/indigo_optional_drivers/ccd_rpi/indigo_ccd_rpi_main.c
index 33ff25b519af..c3abd161cb96 100644
--- a/indigo_optional_drivers/ccd_rpi/indigo_ccd_rpi_main.c
+++ b/indigo_optional_drivers/ccd_rpi/indigo_ccd_rpi_main.c
@@ -30,7 +30,7 @@ int main(int argc, const char *argv[]) {
   indigo_main_argc = argc;
   indigo_main_argv = argv;
 
-  indigo_client *protocol_adapter = indigo_xml_device_adapter(0, 1);
+  indigo_client *protocol_adapter = indigo_xml_device_adapter(indigo_stdin_handle, indigo_stdout_handle);
   indigo_enable_blob_mode_record *record =
       (indigo_enable_blob_mode_record *)malloc(
           sizeof(indigo_enable_blob_mode_record));
-- 
2.55.0

